src/client/Microsoft.Identity.Client/Platforms/iOS/iOSTokenCacheAccessor.cs

```diff
@@ -147,6 +146,25 @@ public IEnumerable<MsalAccountCacheItem> GetAllAccounts()
                 .ToList();
         }
 
+        internal string GetBrokerApplicationToken(string clientId)
+        {
+            var queryRecord = new SecRecord(SecKind.GenericPassword)
```

Comment:
Changes are ok, but it's not clear (i.e. please add some comments) 

a. what is the app token
b. what is the consequence of not having an app token. Today you treat keychaing access errors silently, so it is optional? I'd recommend we make it mandatory - it's better to fail 

KeyChain access has been notoriously bad - I think we need more logging here.

Reply:
iOS team is not throwing if they have a [keychain error w/the apptoken](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/523/files), they are just logging the info here. the consequence is a possible extra consent screen depending on if other parameters are not met. i don't think throwing makes sense here.
I have added some comments
we are logging the result of trying to save the application token, so we'll know if it's a duplicate or something else happened.

---
In reply to: [339137282](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1475#discussion_r339137282) [](ancestors = 339137282)

Reply:
Fair enough.